src/librustc/mir/interpret/mod.rs

```diff
@@ -117,6 +117,10 @@ impl<'tcx> MemoryPointer {
         MemoryPointer { alloc_id, offset }
     }
 
+    pub fn zero(alloc_id: AllocId) -> Self {
```

Comment:
Shouldn't this be a `From` impl?

Reply:
I considered that. But i think being explicit is better. I don't feel too strongly about that though, so if you want I'll change it

Reply:
I think `zero` is misleading, I initially assumed it'd be a null pointer.

Reply:
Also, random thought: maybe `MemoryPointer should be `AllocPointer` or `AbstractPointer`?

Reply:
We could just name it Pointer and rename the current Pointer struct to PrimvalPointer

Reply:
That seems worse. Maybe `enum Scalar { Undef, Bits(u128), Ptr(Pointer) }`, with `enum Value { Scalar(Scalar), ScalarPair(Scalar, Scalar), ByRef(Pointer, Align) }`?

Reply:
What about the current `struct Pointer(Primval)` type? It's just a wrapper around a PrimVal, but helps us not confuse normal PrimVals with those whose type is a pointer type.

Reply:
The type doesn't really matter, the value does (unless there are pointer-specific methods on the wrapper?). And I prefer `Scalar` over `Primval`, but I also forgot the latter is really a thing.